packages/bpk-component-split-input/src/form-test.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable no-await-in-loop */
```

Comment:
Tried so hard to avoid the await in loop (with making Matthew frustrated too) but using promises made this simple code be too unnecessary complicated, open to ideas. :) 
One of the ways we were trying:
`async function ourPromise(inputList: HTMLInputElement[]) {
  const input = inputList.pop()
  return new Promise(async (resolve, reject) => {
    await userEvent.click(input as HTMLInputElement);
    await userEvent.keyboard('1');
    resolve(inputList);
  }).then(ourPromise);
}

await ourPromise(splitInputs as HTMLInputElement[]);`

Reply:
Then I'd have to do it twice, isn't that uglier?